opti/tools/reduce.py

```diff
@@ -0,0 +1,271 @@
+    B[M-1:, :M-1] = -np.eye(M-1)
+
+    B[:M-1,-1] = inputs.bounds.loc["max"]
+    B[M-1:,-1] = -inputs.bounds.loc["min"]
+
+    #eliminate columns with pivot element
+    for i in range(len(pivots)):
+        p = pivots[i]
+        B[p,:] -= A_aug_rref[i,:]
+        B[p+M-1,:] += A_aug_rref[i,:]
+
+
+    #build up reduced problem
+    _inputs = categoricalInputs
+    for i in range(len(inputs)):
+        #add all inputs that were not eliminated
+        if i not in pivots:
+            _inputs.append(inputs[names[i]])
+    _inputs = Parameters(_inputs)
+
+    _constraints = otherConstraints
+    for i in pivots:
+        ind = np.where(B[i,:-1] != 0)[0]
+        if len(ind)>0:
+            c = LinearEquality(names=list(names[ind]),lhs=B[i,ind],rhs=B[i,-1])
+            _constraints.append(c)
+        else:
+            if B[i,-1] < -1e-16:
+                raise Warning("There is no solution in the domain of the variables "
+                "that fulfills the constraints.")
+        
+        ind = np.where(B[i+M-1,:-1] != 0)[0]
+        if len(ind)>0:
+            c = LinearEquality(names=list(names[ind]),lhs=B[i+M-1,ind],rhs=B[i+M-1,-1])
+            _constraints.append(c)
+        else:
+            if B[i+M-1,-1] < -1e-16:
+                raise Warning("There is no solution in the domain of the variables "
+                "that fulfills the constraints.")
+    _constraints = Constraints(_constraints)
+
+    equalities = []
+    for i in range(len(pivots)):
+        name = names[pivots[i]]
+        lhs = ""
+        
+        for j in range(len(names)-1):
+            if (A_aug_rref[i,j] != 0 and j != pivots[i]):
+                lhs += "("+str(-A_aug_rref[i,j]) + ") * " + names[j] + " + "
+            
+        if A_aug_rref[i,-1] != 0:
+            lhs += "("+str(A_aug_rref[i,-1])+")"
+        else:
+            lhs = lhs[:-1]
+        
+        if lhs=="":
+            lhs = "0"
+
+        equalities.append([name, lhs])
+
+    _data = problem.data
+    drop = []
+    if _data is not None:
+        for col in _data.columns:
+            if col not in _inputs.names and col not in problem.outputs.names:
+                drop.append(col)
+        _data = _data.drop(columns=drop)
+
+    #TODO
+    _models = problem.models
+    if _models is not None:
+        pass
+
+    #TODO
+    _f = None
+    if 'f' in list(vars(problem).keys()):
+        _f = problem.f
+        if _f is not None:
+            pass
+
+    _problem = Problem(
+        inputs = _inputs,
+        outputs = problem.outputs,
+        objectives = problem.objectives,
+        constraints = _constraints,                       #BEI DIESEN PUNKTEN NOCH EINMAL DAVID FRAGEN:
+        f = _f,                                           #Einfach daten erweitern, oder?
+        models = _models,                                 #Einfach daten erweitern, oder?
+        data = _data,                                     #droppe alle spalten, deren Variable eliminiert wurde, richtig so?
```

Comment:
The eliminated variables don't need to be dropped from the data.

Reply:
Understood, I thought this would cause problems when `check_data()` is called, but it does not.